(window["webpackJsonp"]=window["webpackJsonp"]||[]).push([["tema1"],{"0083":function(e,a,t){e.exports=t.p+"img/img-25.810cf6c7.png"},"0f30":function(e,a,t){e.exports=t.p+"img/img-14.bc60d9c1.png"},"11e7":function(e,a,t){e.exports=t.p+"img/img-33.36eaa29f.png"},1246:function(e,a,t){e.exports=t.p+"img/img-11.6ecf47e3.png"},"1d21":function(e,a,t){e.exports=t.p+"img/img-8.f24f5140.png"},2391:function(e,a,t){e.exports=t.p+"img/material-complementario.06dbfac1.svg"},3244:function(e,a,t){e.exports=t.p+"img/img-21.2e56e99e.png"},"33d9":function(e,a,t){e.exports=t.p+"img/img-13.4b91b3dc.png"},"3a07":function(e,a,t){e.exports=t.p+"img/img-32.26b1fe89.svg"},"52b3":function(e,a,t){e.exports=t.p+"img/img-6.47ea9595.png"},"5cab":function(e,a,t){e.exports=t.p+"img/img-10.d40dd241.png"},"5e35":function(e,a,t){e.exports=t.p+"img/youtube-icon.0e5f1361.svg"},"5f56":function(e,a,t){e.exports=t.p+"img/img-9.e3a6a035.png"},"643a":function(e,a,t){e.exports=t.p+"img/pdf-icon-square.6a47789e.svg"},"664b":function(e,a,t){e.exports=t.p+"img/img-34.7ec3c64d.png"},6984:function(e,a,t){e.exports=t.p+"img/img-24.d2ac1882.png"},"741e":function(e,a,t){e.exports=t.p+"img/img-29.681c9f5d.svg"},8221:function(e,a,t){e.exports=t.p+"img/img-18.8866f0ef.png"},8689:function(e,a,t){e.exports=t.p+"media/Programación_orientada_a_objetos.555a92a2.mp3"},"8f56":function(e,a,t){e.exports=t.p+"img/img-26.a5eddeec.png"},9051:function(e,a,t){e.exports=t.p+"img/img-5.aaaa2ef1.svg"},9229:function(e,a,t){e.exports=t.p+"img/img-19.1320111c.png"},"973e":function(e,a,t){e.exports=t.p+"img/img-16.361605b3.png"},aee4:function(e,a,t){e.exports=t.p+"img/img-4.84625c02.png"},b14e:function(e,a,t){"use strict";t.r(a);var s=function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"curso-main-container pb-3"},[s("BannerInterno",{attrs:{subTitulo:"1. Manejo de excepciones y eventos"}}),s("div",{staticClass:"container tarjeta tarjeta--blanca p-4 p-md-5 overflow-hidden"},[e._m(0),e._m(1),e._m(2),e._m(3),e._m(4),e._m(5),s("h2",[e._v("Excepciones verificadas y no verificadas")]),e._m(6),e._m(7),e._m(8),e._m(9),s("h2",[e._v("Jerarquía de clases de excepción")]),e._m(10),e._m(11),e._m(12),e._m(13),e._m(14),s("h2",[e._v("Declaración y propagación de excepciones")]),e._m(15),s("div",{staticClass:"row position-relative mb-5 py-4 bg-fondo-slider bg-full-width"},[s("div",{staticClass:"px-4 px-md-5 py-md-3"},[s("SlyderA",{staticClass:"bg-white",attrs:{tipo:"b"}},[s("div",{staticClass:"row p-4 px-md-5 py-lg-5 justify-content-between"},[s("div",{staticClass:"col-md-12 col-lg-7 mb-4 mb-md-0 d-flex flex-column justify-content-center"},[s("p",{staticClass:"text-bold"},[e._v("Try")]),s("p",[e._v("Se utiliza para encapsular el código que puede lanzar una excepción. Si se produce una excepción dentro del bloque "),s("i",[e._v("try")]),e._v(", el flujo del programa se transfiere al bloque catch correspondiente.")])]),s("div",{staticClass:"d-none d-lg-block col-md-6 col-lg-5",attrs:{"data-aos":"zoom-in"}},[s("figure",[s("img",{attrs:{src:t("973e"),alt:"Texto que describa la imagen"}})])])]),s("div",{staticClass:"row p-4 px-md-5 py-lg-5 justify-content-between"},[s("div",{staticClass:"col-md-12 col-lg-7 mb-4 mb-md-0 d-flex flex-column justify-content-center"},[s("p",{staticClass:"text-bold"},[e._v("Catch")]),s("p",[e._v("Se utiliza para manejar las excepciones lanzadas en el bloque "),s("i",[e._v("try")]),e._v(". Se especifica el tipo de excepción que se desea capturar y se proporciona el código para manejar esa excepción.")])]),s("div",{staticClass:"d-none d-lg-block col-md-6 col-lg-5",attrs:{"data-aos":"zoom-in"}},[s("figure",[s("img",{attrs:{src:t("d9e3"),alt:"Texto que describa la imagen"}})])])]),s("div",{staticClass:"row p-4 px-md-5 py-lg-5 justify-content-between"},[s("div",{staticClass:"col-md-12 col-lg-7 mb-4 mb-md-0 d-flex flex-column justify-content-center"},[s("p",{staticClass:"text-bold"},[e._v("Finally")]),s("p",[e._v("Se utiliza para especificar un bloque de código que se ejecutará siempre, independientemente de si se produjo una excepción o no. Es útil para liberar recursos, como cerrar archivos o conexiones a bases de datos.")])]),s("div",{staticClass:"d-none d-lg-block col-md-6 col-lg-5",attrs:{"data-aos":"zoom-in"}},[s("figure",[s("img",{attrs:{src:t("8221"),alt:"Texto que describa la imagen"}})])])])])],1)]),s("p",{staticClass:"mb-4"},[e._v("A continuación, se presenta una tabla con los términos más relevantes y su función en el manejo de excepciones.")]),e._m(16),e._m(17),e._m(18),e._m(19),s("h2",[e._v("Creación de excepciones personalizadas")]),s("div",{staticClass:"row mb-4"},[s("div",{staticClass:"col-12 col-xl-7"},[s("p",{staticClass:"mb-4"},[e._v("A continuación, se presenta un recurso educativo que aborda la creación de excepciones personalizadas en programación orientada a objetos, explicando su propósito, implementación y beneficios para un manejo de errores más claro y específico: ")]),s("TarjetaAudio",{staticClass:"color-secundario bg-white mb-3",attrs:{texto:"Programación orientada a objetos",tiempo:"",audio:t("8689")},on:{"audio-hover":function(a){e.mostrarIndicadorTarjetaAudio=!1}}})],1),e._m(20)]),e._m(21),e._m(22),e._m(23),s("h2",[e._v("Buenas prácticas para el manejo de errores")]),s("p",{staticClass:"mb-4"},[e._v("El manejo técnico de errores es fundamental para crear aplicaciones robustas y fáciles de mantener. A continuación, se presentan algunas buenas prácticas para el manejo de errores en la programación orientada a objetos:")]),s("div",{staticClass:"row bg-fondo-1 bg-full-width py-4 mb-4"},[s("div",{staticClass:"px-5"},[s("div",{staticClass:"row"},[e._m(24),s("div",{staticClass:"col-12 col-xl-8"},[s("SlyderF",{attrs:{columnas:"col-12 col-lg-6 py-4 px-3"}},[s("div",{staticClass:"custom-image-card d-flex flex-column p-4 py-md-5"},[s("img",{staticClass:"mx-auto mb-4",staticStyle:{width:"80px"},attrs:{src:t("b720")}}),s("p",{staticClass:"text-bold mb-4 text-center"},[e._v("Utilizar excepciones para situaciones excepcionales")]),s("p",{staticClass:"px-1 text-small mb-0 text-center"},[e._v("Las excepciones deben utilizarse para manejar situaciones excepcionales o errores, no para controlar el flujo normal del programa")])]),s("div",{staticClass:"custom-image-card d-flex flex-column p-4 py-md-5"},[s("img",{staticClass:"mx-auto mb-4",staticStyle:{width:"80px"},attrs:{src:t("bd3d")}}),s("p",{staticClass:"text-bold mb-4 text-center"},[e._v("Ser específico con las excepciones")]),s("p",{staticClass:"px-1 text-small mb-0 text-center"},[e._v("Utilizar excepciones específicas que representen claramente el tipo de error o situación excepcional que se está manejando")])]),s("div",{staticClass:"custom-image-card d-flex flex-column p-4 py-md-5"},[s("img",{staticClass:"mx-auto mb-4",staticStyle:{width:"80px"},attrs:{src:t("741e")}}),s("p",{staticClass:"text-bold mb-4 text-center"},[e._v("Proporcionar mensajes de error informativos")]),s("p",{staticClass:"px-1 text-small mb-0 text-center"},[e._v("Los mensajes de error deben ser claros, concisos y proporcionar información relevante para el diagnóstico y solución del problema")])]),s("div",{staticClass:"custom-image-card d-flex flex-column p-4 py-md-5"},[s("img",{staticClass:"mx-auto mb-4",staticStyle:{width:"80px"},attrs:{src:t("b923")}}),s("p",{staticClass:"text-bold mb-4 text-center"},[e._v("Evitar el uso excesivo de bloques try-catch")]),s("p",{staticClass:"px-1 text-small mb-0 text-center"},[e._v("Solo cuando sea necesario y evitar anidar múltiples bloques try-catch, ya que puede dificultar la legibilidad del código")])]),s("div",{staticClass:"custom-image-card d-flex flex-column p-4 py-md-5"},[s("img",{staticClass:"mx-auto mb-4",staticStyle:{width:"80px"},attrs:{src:t("ef84")}}),s("p",{staticClass:"text-bold mb-4 text-center"},[e._v("Liberar recursos en el bloque finally")]),s("p",{staticClass:"px-1 text-small mb-0 text-center"},[e._v("Utilizar el bloque finally para liberar recursos, como cerrar archivos o conexiones a bases de datos, independientemente de si se produjo una excepción o no.")])]),s("div",{staticClass:"custom-image-card d-flex flex-column p-4 py-md-5"},[s("img",{staticClass:"mx-auto mb-4",staticStyle:{width:"80px"},attrs:{src:t("3a07")}}),s("p",{staticClass:"text-bold mb-4 text-center"},[e._v("No ignorar las excepciones")]),s("p",{staticClass:"px-1 text-small mb-0 text-center"},[e._v("Evitar bloques catch vacíos o que simplemente registren el error sin realizar ninguna acción. Siempre se debe manejar adecuadamente las excepciones o propagarlas si no se pueden manejar en el nivel actual")])])])],1)])])]),e._m(25),e._m(26),e._m(27),s("div",{staticClass:"bg-full-width border-top-9-azul"},[s("div",{staticClass:"px-5 pt-5 pb-0"},[s("h2",[e._v("Material complementario")]),s("div",{staticClass:"row"},[s("div",{staticClass:"col-12 col-md-6 col-lg-8"},[s("p",{staticClass:"d-flex my-4"},[s("img",{staticClass:"me-3 mb-auto mt-1",style:{"max-width":"16px"},attrs:{src:t("f021")}}),s("a",{staticClass:"text-small text-primary",attrs:{target:"_blank",href:"https://elibro.net/es/ereader/tecnologicadeloriente/106519?page=1"}},[e._v("Ceballos Sierra, F. J. (2018). Programación orientada a objetos con C++: (5 ed.). RA-MA Editorial")])]),s("p",{staticClass:"d-flex my-4"},[s("img",{staticClass:"me-3 mb-auto mt-1",style:{"max-width":"16px"},attrs:{src:t("f021")}}),s("a",{staticClass:"text-small text-primary",attrs:{target:"_blank",href:"https://elibro.net/es/ereader/tecnologicadeloriente/39461?page=1"}},[e._v("López Goytia, J. L. (2015). Programación orientada a objetos C++ y Java: un acercamiento interdisciplinario: ( ed.). Grupo Editorial Patria")])]),s("p",{staticClass:"d-flex my-4"},[s("img",{staticClass:"me-3 mb-auto mt-1",style:{"max-width":"16px"},attrs:{src:t("f021")}}),s("a",{staticClass:"text-small text-primary",attrs:{target:"_blank",href:"https://elibro.net/es/ereader/tecnologicadeloriente/106461?page=1"}},[e._v("Moreno Pérez, J. C. (2015). Programación orientada a objetos: ( ed.). RA-MA Editorial")])]),s("p",{staticClass:"d-flex my-4"},[s("img",{staticClass:"me-3 mb-auto mt-1",style:{"max-width":"16px"},attrs:{src:t("f021")}}),s("a",{staticClass:"text-small text-primary",attrs:{target:"_blank",href:"https://elibro.net/es/ereader/tecnologicadeloriente/70431?page=1"}},[e._v("Oviedo Regino, E. M. (2015). Lógica de programación orientada a objetos: ( ed.). Ecoe Ediciones")])]),s("p",{staticClass:"d-flex my-4"},[s("img",{staticClass:"me-3 mb-auto mt-1",style:{"max-width":"16px"},attrs:{src:t("f021")}}),s("a",{staticClass:"text-small text-primary",attrs:{target:"_blank",href:"https://elibro.net/es/ereader/tecnologicadeloriente/34869?page=9"}},[e._v("Ruiz Rodríguez, R. (2009). Fundamentos de la programación orientada a objetos: una aplicación a las estructuras de datos en Java: ( ed.). El Cid Editor")])]),s("p",{staticClass:"d-flex my-4"},[s("img",{staticClass:"me-3 mb-auto mt-1",style:{"max-width":"16px"},attrs:{src:t("5e35")}}),s("a",{staticClass:"text-small text-primary",attrs:{target:"_blank",href:"https://youtu.be/Wnn6RZ4opfA"}},[e._v("[Vídeo] Cruz, R. (2020, 28 de octubre). Manejo de Excepciones")])]),s("p",{staticClass:"d-flex my-4"},[s("img",{staticClass:"me-3 mb-auto mt-1",style:{"max-width":"16px"},attrs:{src:t("5e35")}}),s("a",{staticClass:"text-small text-primary",attrs:{target:"_blank",href:"https://youtu.be/7XeQBBcaFZQ"}},[e._v("[Vídeo] Castillo, L. (2020, 16 de agosto). Manejo de excepciones en Java (try-catch)")])])]),e._m(28)])])]),s("p")])],1)},i=[function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"row mb-4"},[s("div",{staticClass:"col-12 col-xl-9"},[s("span",[e._v("En la programación orientada a objetos, el dominio de excepciones y eventos es fundamental para crear aplicaciones robustas y fáciles de mantener. Las excepciones permiten controlar situaciones inesperadas o errores durante la ejecución del programa, evitando así que la aplicación se detenga abruptamente, brindando información valiosa para el diagnóstico y solución de problemas.")])]),s("div",{staticClass:"col-3 d-none d-xl-block"},[s("img",{attrs:{src:t("b640"),alt:"Texto que describa la imagen"}})])])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"row bg-azul-oscuro bg-full-width py-4 mb-4"},[s("div",{staticClass:"px-5 pb-md-3 py-4 d-flex"},[s("div",{staticClass:"d-none d-xl-block col-xl-4 mx-auto mt-4 mt-xl-0",attrs:{"data-aos":"zoom-in"}},[s("img",{staticClass:"h-100",attrs:{src:t("aee4"),alt:"Texto que describa la imagen"}})]),s("div",{staticClass:"col-12 col-xl-8 d-flex ps-4 ps-xl-5 flex-column"},[s("h2",{staticClass:"mb-4 pb-2 me-auto text-white"},[e._v("Ejemplo ")]),s("p",{staticClass:"mb-4 text-white",attrs:{"data-aos":"fade-up"}},[e._v("Para conocer un ejemplo práctico de excepciones le invitamos a ver el siguiente vídeo.")]),s("p",{staticClass:"mb-4 text-white",attrs:{"data-aos":"fade-up"}},[e._v("Manejo de Excepciones en POO explicado en 2 minutos. ")]),s("p",{staticClass:"mb-5 text-white",attrs:{"data-aos":"fade-up"}},[e._v("Además, a través del siguiente enlace podrá interactuar con el código, crear sus propias excepciones o modificar sus atributos para obtener nuevos resultados. ")]),s("a",{staticClass:"d-flex me-auto w-fit bg-white box-shadow cursor-pointer",attrs:{"data-aos":"fade-up",href:"https://www.programiz.com/online-compiler/1gfDwEarOuYWz ",target:"_blank"}},[s("img",{staticClass:"h-100",staticStyle:{width:"48px"},attrs:{src:t("bbe3"),alt:"Texto que describa la imagen"}}),s("p",{staticClass:"text-small fs-14px my-auto px-2"},[s("strong",[e._v("Video.")]),e._v(" Manejo de Excepciones en POO")])])])])])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"row mb-4"},[s("div",{staticClass:"col-12 col-xl-8"},[s("p",{staticClass:"mb-4"},[e._v("Por otro lado, los eventos son una parte esencial de las interfaces gráficas de usuario (GUI) que permiten a las aplicaciones responder a las acciones del usuario, como clics de ratón o pulsaciones de teclas. El manejo de eventos se basa en el diseño orientado a eventos (Event-Driven Programming), donde el flujo del programa está determinado por los eventos generados por el usuario o el sistema.")]),s("div",{staticClass:"d-flex bg-verde p-4 flex-column flex-md-row mb-4",attrs:{"data-aos":"fade-up"}},[s("img",{staticClass:"me-3",staticStyle:{width:"90px"},attrs:{src:t("9051"),alt:"Texto descriptivo de la imagen"}}),s("div",{staticClass:"my-auto p-4 p-xl-0"},[s("p",{staticClass:"pe-xl-3 mb-0"},[e._v("En esta unidad, se explorarán en detalle los conceptos relacionados con el manejo de excepciones y eventos en el contexto de la programación orientada a objetos. Se abordarán los diferentes tipos de excepciones, la jerarquía de clases de excepción, la declaración y propagación de excepciones, así como la creación de excepciones personalizadas y buenas prácticas para el manejo de errores.")])])])]),s("div",{staticClass:"col-4 d-none d-xl-block"},[s("img",{attrs:{src:t("52b3"),alt:"Texto que describa la imagen"}})])])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"row mb-4"},[s("div",{staticClass:"col-4 d-none d-xl-block"},[s("img",{attrs:{src:t("fd6e"),alt:"Texto que describa la imagen"}})]),s("div",{staticClass:"col-12 col-xl-8"},[s("p",{staticClass:"mb-4"},[e._v("Además, se profundizará en el manejo de eventos en interfaces gráficas, incluyendo las clases de eventos y "),s("i",[e._v("listeners")]),e._v(", la asociación de eventos a componentes, la implementación de interfaces de escucha y el diseño orientado a eventos.")]),s("div",{staticClass:"bg-morado p-4 mb-4",attrs:{"data-aos":"fade-up"}},[s("span",[e._v("Finalmente, se abordarán las pruebas y depuración de excepciones, con énfasis en técnicas básicas de depuración, el uso de herramientas del entorno de desarrollo, la identificación y corrección de errores comunes, el registro de errores y la integración de pruebas unitarias para validar el manejo de errores.")])]),s("span",[e._v("Al finalizar esta unidad, se tendrá un conocimiento sólido sobre el manejo de excepciones y eventos en la programación orientada a objetos, permitiendo desarrollar aplicaciones más robustas, fáciles de mantener y con una mejor experiencia de usuario.")])])])},function(){var e=this,a=e.$createElement,t=e._self._c||a;return t("div",{staticClass:"titulo-segundo",attrs:{id:"t_1_1","data-aos":"flip-up"}},[t("div",{staticClass:"d-flex"},[t("span",{staticClass:"me-3"},[e._v("1.1")]),t("h2",[e._v("Tipos de excepciones")])])])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"row mb-4"},[s("div",{staticClass:"col-12 col-xl-8"},[s("p",{staticClass:"mb-4"},[e._v("En la programación orientada a objetos, las excepciones son un mecanismo para manejar situaciones excepcionales o errores que pueden ocurrir durante la ejecución de un programa. Las excepciones permiten separar el código de manejo de errores del flujo normal del programa, lo que facilita la legibilidad y el mantenimiento del código.")]),s("div",{staticClass:"bg-azul p-4",attrs:{"data-aos":"fade-up"}},[s("span",[e._v("En este apartado, se explorarán los diferentes tipos de excepciones, incluyendo las excepciones verificadas y no verificadas, la jerarquía de clases de excepción, la declaración y propagación de excepciones, la creación de excepciones personalizadas y las buenas prácticas para el manejo de errores.")])])]),s("div",{staticClass:"col-4 d-none d-xl-block"},[s("img",{attrs:{src:t("1d21"),alt:"Texto que describa la imagen"}})])])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"row mb-4"},[s("div",{staticClass:"col-4 d-none d-xl-block"},[s("img",{attrs:{src:t("5f56"),alt:"Texto que describa la imagen"}})]),s("div",{staticClass:"col-12 col-xl-8"},[s("p",{staticClass:"mb-4"},[e._v("Las excepciones se clasifican en dos categorías principales: excepciones verificadas ("),s("i",[e._v("checked exceptions")]),e._v(") y excepciones no verificadas ("),s("i",[e._v("unchecked exceptions")]),e._v(").")]),s("div",{staticClass:"bg-morado p-4",attrs:{"data-aos":"fade-up"}},[s("span",[e._v("Las excepciones verificadas son aquellas que deben ser declaradas en la firma del método o manejadas dentro del mismo. El compilador verifica que estas excepciones sean tratadas adecuadamente, ya sea mediante un bloque try-catch o declarándolas en la cláusula throws. Ejemplos de excepciones verificadas en Java son IOException, SQLException y ClassNotFoundException.")])])])])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"row mb-4"},[s("div",{staticClass:"col-12 col-xl-8"},[s("p",{staticClass:"mb-4"},[e._v("Por otro lado, las excepciones no verificadas no requieren ser declaradas ni manejadas explícitamente. Estas excepciones heredan de la clase RuntimeException o Error y son generalmente causadas por errores de programación, como acceso a índices fuera de los límites de un arreglo (ArrayIndexOutOfBoundsException) o división por cero (ArithmeticException).")]),s("div",{staticClass:"bg-amarillo p-4 mb-4",attrs:{"data-aos":"fade-up"}},[s("span",[e._v("Según Ceballos Sierra (2018), las excepciones verificadas obligan a que el programador tome una decisión sobre cómo manejar la excepción, mientras que las excepciones no verificadas no requieren una acción explícita del programador.")])]),s("span",[e._v("En el contexto de la ingeniería de software, es importante distinguir entre excepciones verificadas y no verificadas al diseñar y desarrollar aplicaciones. Las excepciones verificadas se utilizan para situaciones excepcionales que pueden ocurrir y que deben ser manejadas explícitamente, como errores de entrada/salida o de base de datos. Por otro lado, las excepciones no verificadas se utilizan para errores de programación y situaciones inesperadas que no siempre pueden ser anticipadas.")])]),s("div",{staticClass:"col-4 d-none d-xl-block"},[s("img",{attrs:{src:t("5cab"),alt:"Texto que describa la imagen"}})])])},function(){var e=this,a=e.$createElement,t=e._self._c||a;return t("p",{staticClass:"mb-4"},[t("b",[e._v("Figura 1.")]),e._v(" Ejemplo de uso de excepciones.")])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"row mb-4 bg-azul py-4"},[s("div",{staticClass:"col-10 col-md-8 col-lg-6 mx-auto"},[s("img",{attrs:{src:t("1246"),alt:"Texto que describa la imagen"}})])])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"row mb-4"},[s("div",{staticClass:"col-4 d-none d-xl-block"},[s("img",{attrs:{src:t("c594"),alt:"Texto que describa la imagen"}})]),s("div",{staticClass:"col-12 col-xl-8"},[s("p",{staticClass:"mb-4"},[e._v("En la programación orientada a objetos, las excepciones se organizan en una jerarquía de clases. En Java, la clase base para todas las excepciones y errores es "),s("i",[e._v("Throwable")]),e._v(". A partir de "),s("i",[e._v("Throwable")]),e._v(", se derivan dos subclases principales: Exception y Error.")]),s("div",{staticClass:"bg-morado p-4",attrs:{"data-aos":"fade-up"}},[s("span",[e._v("La clase Exception representa las excepciones que pueden ser manejadas por el programa. Dentro de Exception, se encuentra la subclase RuntimeException, que engloba las excepciones no verificadas. Todas las demás excepciones que heredan directamente de Exception son consideradas excepciones verificadas.")])])])])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"row mb-4"},[s("div",{staticClass:"col-12 col-xl-8"},[s("p",{staticClass:"mb-4"},[e._v("Por otro lado, la clase Error representa errores graves que generalmente no pueden ser manejados por el programa, como errores de la máquina virtual de Java (OutOfMemoryError) o errores de sintaxis en el código fuente (VerifyError).")]),s("div",{staticClass:"bg-verde p-4 mb-4",attrs:{"data-aos":"fade-up"}},[s("span",[e._v("De acuerdo con Moreno Pérez (2015), la jerarquía de clases de excepción permite al programador crear excepciones personalizadas que se ajusten a las necesidades específicas de la aplicación.")])]),s("span",[e._v("En el desarrollo de software, es común crear excepciones personalizadas que hereden de Exception o RuntimeException para representar situaciones excepcionales específicas del dominio de la aplicación. Esto permite un manejo de errores más granular y facilita la comprensión y el mantenimiento del código. ")])]),s("div",{staticClass:"col-4 d-none d-xl-block"},[s("img",{attrs:{src:t("33d9"),alt:"Texto que describa la imagen"}})])])},function(){var e=this,a=e.$createElement,t=e._self._c||a;return t("p",{staticClass:"mb-4"},[t("b",[e._v("Figura 2.")]),e._v(" Ejemplo de jerarquía de clases de excepción.")])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"row mb-4 bg-azul py-4"},[s("div",{staticClass:"col-10 col-md-8 col-lg-5 mx-auto"},[s("img",{attrs:{src:t("0f30"),alt:"Texto que describa la imagen"}})])])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"row bg-azul-oscuro bg-full-width py-4 mb-4"},[s("div",{staticClass:"px-5 pb-md-3 py-4 d-flex"},[s("div",{staticClass:"d-none d-xl-block col-xl-4 mx-auto mt-4 mt-xl-0",attrs:{"data-aos":"zoom-in"}},[s("img",{staticClass:"h-100",attrs:{src:t("ce4b"),alt:"Texto que describa la imagen"}})]),s("div",{staticClass:"col-12 col-xl-8 d-flex ps-4 ps-xl-5 align-items-center flex-column"},[s("h2",{staticClass:"mb-4 pb-2 me-auto text-white"},[e._v("Manejo de excepciones")]),s("p",{staticClass:"mb-5 text-white",attrs:{"data-aos":"fade-up"}},[e._v("Para ampliar el concepto sobre el manejo de excepciones, le invitamos a ver el siguiente vídeo. ")]),s("a",{staticClass:"d-flex me-auto w-fit bg-white box-shadow cursor-pointer",attrs:{"data-aos":"fade-up",href:"https://youtu.be/Wnn6RZ4opfA ",target:"_blank"}},[s("img",{staticClass:"h-100",staticStyle:{width:"48px"},attrs:{src:t("643a"),alt:"Texto que describa la imagen"}}),s("p",{staticClass:"text-small fs-14px my-auto px-2"},[s("strong",[e._v("Video.")]),e._v(" Manejo de Excepciones.")])])])])])},function(){var e=this,a=e.$createElement,t=e._self._c||a;return t("p",{staticClass:"mb-4"},[e._v("En la programación orientada a objetos, las excepciones se declaran y propagan utilizando las palabras clave: "),t("i",[e._v("try, catch, finally, throw y throws")]),e._v(".")])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"row bg-fondo-1 bg-full-width py-4 mb-4"},[s("div",{staticClass:"px-5"},[s("div",{staticClass:"row mb-4"},[s("div",{staticClass:"col-10 col-md-8 col-lg-6 mb-4 mx-auto",attrs:{"data-aos":"fade-right"}},[s("div",{staticClass:"custom-image-card rounded-8px"},[s("img",{staticClass:"mb-4",attrs:{src:t("9229"),alt:"Texto que describa la imagen"}}),s("div",{staticClass:"mb-4 px-4"},[s("p",{staticClass:"text-bold text-center"},[e._v("Throw")]),s("p",{staticClass:"text-small px-4 text-center pb-2"},[e._v("Se utiliza para lanzar una excepción explícitamente desde el código. Esto permite al programador señalar situaciones excepcionales y transferir el control a un bloque catch adecuado.")])])])]),s("div",{staticClass:"col-10 col-md-8 col-lg-6 mb-4 mx-auto",attrs:{"data-aos":"fade-left"}},[s("div",{staticClass:"custom-image-card rounded-8px"},[s("img",{staticClass:"mb-4",attrs:{src:t("efe8"),alt:"Texto que describa la imagen"}}),s("div",{staticClass:"mb-4 px-4"},[s("p",{staticClass:"text-bold text-center"},[e._v("Throws")]),s("p",{staticClass:"text-small px-4 text-center pb-2"},[e._v("Se utiliza en la firma de un método para declarar las excepciones verificadas que el método puede lanzar. Esto obliga a que quien llame al método maneje esas excepciones o las declare a su vez.")])])])])])])])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"row mb-4"},[s("div",{staticClass:"col-12 col-xl-8"},[s("p",{staticClass:"mb-4"},[e._v("López Goytia (2015), considera que el uso adecuado de "),s("i",[e._v("try, catch, finally, throw y throws")]),e._v(" permite un manejo estructurado y eficiente de las excepciones en el código.")]),s("div",{staticClass:"bg-amarillo p-4",attrs:{"data-aos":"fade-up"}},[s("span",[e._v("En el desarrollo de software, es importante utilizar correctamente estos mecanismos para manejar las excepciones de manera efectiva. La declaración o propagación adecuada de las excepciones ayuda a localizar y solucionar errores, además de mejorar la legibilidad y el mantenimiento del código.")])])]),s("div",{staticClass:"col-4 d-none d-xl-block"},[s("img",{attrs:{src:t("3244"),alt:"Texto que describa la imagen"}})])])},function(){var e=this,a=e.$createElement,t=e._self._c||a;return t("p",{staticClass:"mb-4"},[t("b",[e._v("Figura 3.")]),e._v(" Ejemplo de declaración y propagación de excepciones")])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"row mb-4 bg-azul py-4"},[s("div",{staticClass:"col-11 col-md-9 col-lg-7 mx-auto"},[s("img",{attrs:{src:t("f179"),alt:"Texto que describa la imagen"}})])])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"col-5 d-none d-xl-block"},[s("img",{attrs:{src:t("e114"),alt:"Texto que describa la imagen"}})])},function(){var e=this,a=e.$createElement,t=e._self._c||a;return t("p",{staticClass:"mb-4"},[t("b",[e._v("Figura 4.")]),e._v(" Ejemplo de creación de una excepción personalizada")])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"row mb-4 bg-azul py-4"},[s("div",{staticClass:"col-11 col-md-9 col-lg-8 mx-auto"},[s("img",{attrs:{src:t("6984"),alt:"Texto que describa la imagen"}})])])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"row bg-azul-oscuro bg-full-width py-4 mb-4"},[s("div",{staticClass:"px-5 pb-md-3 py-4 d-flex"},[s("div",{staticClass:"d-none d-xl-block col-xl-4 mx-auto mt-4 mt-xl-0",attrs:{"data-aos":"zoom-in"}},[s("img",{staticClass:"h-100",attrs:{src:t("0083"),alt:"Texto que describa la imagen"}})]),s("div",{staticClass:"col-12 col-xl-8 d-flex ps-4 ps-xl-5 align-items-center flex-column"},[s("h2",{staticClass:"mb-4 pb-2 me-auto text-white"},[e._v("Manejo de excepciones en Java (try-catch). ")]),s("p",{staticClass:"mb-5 text-white",attrs:{"data-aos":"fade-up"}},[e._v("Para ampliar el concepto sobre el manejo de excepciones en Java, le invitamos a ver el siguiente vídeo. ")]),s("a",{staticClass:"d-flex me-auto w-fit bg-white box-shadow cursor-pointer",attrs:{"data-aos":"fade-up",href:"https://youtu.be/7XeQBBcaFZQ",target:"_blank"}},[s("img",{staticClass:"h-100",staticStyle:{width:"48px"},attrs:{src:t("643a"),alt:"Texto que describa la imagen"}}),s("p",{staticClass:"text-small fs-14px my-auto px-2"},[s("strong",[e._v("Video.")]),e._v(" Manejo de excepciones en Java (try-catch).")])])])])])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"col-4 d-none d-xl-block"},[s("img",{attrs:{src:t("8f56"),alt:"Texto que describa la imagen"}})])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"row mb-4"},[s("div",{staticClass:"col-12 col-xl-8"},[s("p",{staticClass:"mb-4"},[e._v("Según Ruiz Rodríguez (2009), el manejo adecuado de errores es una parte esencial del desarrollo de "),s("i",[e._v("software")]),e._v(" que contribuye a la creación de aplicaciones más robustas y fáciles de mantener.")]),s("div",{staticClass:"bg-amarillo p-4",attrs:{"data-aos":"fade-up"}},[s("span",[e._v("En el contexto de la ingeniería de "),s("i",[e._v("software")]),e._v(", seguir buenas prácticas para el manejo de errores ayuda a mejorar la calidad del software, facilita el diagnóstico y solución de problemas, y reduce el tiempo y los costos de mantenimiento.")])])]),s("div",{staticClass:"col-4 d-none d-xl-block"},[s("img",{attrs:{src:t("11e7"),alt:"Texto que describa la imagen"}})])])},function(){var e=this,a=e.$createElement,t=e._self._c||a;return t("p",{staticClass:"mb-4"},[t("b",[e._v("Figura 5.")]),e._v(" Ejemplo de buenas prácticas para el manejo de errores")])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"row mb-5 bg-azul py-4"},[s("div",{staticClass:"col-11 col-md-9 col-lg-8 mx-auto"},[s("img",{attrs:{src:t("664b"),alt:"Texto que describa la imagen"}})])])},function(){var e=this,a=e.$createElement,s=e._self._c||a;return s("div",{staticClass:"col-12 col-md-6 col-lg-3"},[s("figure",[s("img",{attrs:{src:t("2391"),alt:"Imagen de material complementario"}})])])}],o={name:"Tema1",mounted:function(){var e=this;this.$nextTick((function(){e.$aosRefresh()}))}},c=o,r=t("2877"),l=Object(r["a"])(c,s,i,!1,null,null,null);a["default"]=l.exports},b640:function(e,a,t){e.exports=t.p+"img/img-3.4e4de2b7.png"},b720:function(e,a,t){e.exports=t.p+"img/img-27.302662e8.svg"},b923:function(e,a,t){e.exports=t.p+"img/img-30.c1458add.svg"},bbe3:function(e,a,t){e.exports=t.p+"img/youtube-icon-square.3b3ed5e4.svg"},bd3d:function(e,a,t){e.exports=t.p+"img/img-28.a8ca7ea2.svg"},c594:function(e,a,t){e.exports=t.p+"img/img-12.17548b33.png"},ce4b:function(e,a,t){e.exports=t.p+"img/img-15.c1ddf9cd.png"},d9e3:function(e,a,t){e.exports=t.p+"img/img-17.9447d030.png"},e114:function(e,a,t){e.exports=t.p+"img/img-23.e39184f7.png"},ef84:function(e,a,t){e.exports=t.p+"img/img-31.03b92dab.svg"},efe8:function(e,a,t){e.exports=t.p+"img/img-20.a09bf717.png"},f021:function(e,a,t){e.exports=t.p+"img/book.b08cb9da.svg"},f179:function(e,a,t){e.exports=t.p+"img/img-22.0d1c2740.png"},fd6e:function(e,a,t){e.exports=t.p+"img/img-7.36cea828.png"}}]);
//# sourceMappingURL=tema1.0fdd1d17.js.map